pkg/proxy/proxy.go

```diff
@@ -262,22 +261,17 @@ func (p *Proxy) ackProxyPort(pp *ProxyPort) error {
 		// in the datapath, but means that the datapath rules may exist even
 		// if the proxy is not currently configured.
 
-		// Remove old rules, if any and for different port
-		if pp.rulesPort != 0 && pp.rulesPort != pp.proxyPort {
-			scopedLog.Infof("Removing old proxy port rules for %s:%d", pp.name, pp.rulesPort)
-			p.datapathUpdater.RemoveProxyRules(pp.rulesPort, pp.ingress, pp.name)
```

Comment:
May be a naive question, but why was this required before and what logic replaces this? Could there be some stale state issue because this is no longer run here, or is it simply moved elsewhere?

(if at all possible, splitting this particular change into another commit and explaining it would help a lot for understanding the implications of the change).

Reply:
Removal of stale TPROXY rules is now done in the iptables package when new TPROXY rules are added. It is needed when the port bound to the proxy is different from the old one. Normally we reuse the old port, but we do support passing the DNS proxy port as a command line option, I recall.